pkgs/by-name/od/odri-masterboard/package.nix

```diff
@@ -0,0 +1,33 @@
+{
+  fetchFromGitHub,
+  fetchgit,
+  stdenv,
+  cmake,
+  yaml-cpp,
+  eigen,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "odri";
+  version = "1.0.1";
+
+  src = fetchgit {
+    url = "https://github.com/open-dynamic-robot-initiative/master-board";
+    # repo = "odri_control_interface";
+    rev = "v${version}";
+    # sha256 = "sha256-jeZFgAW8C+UCeW+TtV1fNJZR7yuWcpiQnepsbkNanPg=";
+    hash = "sha256-8Yw79we53PE9HHu2j95DR4PJi/MZVtunpFA2at7+4Uk=";
+    sparseCheckout = [
+      "sdk/master_board_sdk"
+    ];
+    fetchSubmodules = true;
+  };
+
+  prePatch = ''
+    cd sdk/master_board_sdk
+  '';
```

Comment:
you can check https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-sourceRoot for this